### PR TITLE
Adds Assistant Borg Module

### PR DESCRIPTION
**Pull request by IndieanaJones**
## About The Pull Request

This PR adds a replacement cyborg module to the game which uses the standard borg sprites, the assistant borg.  The assistant borg intends to fill two niches:

- A borg subtype for new players to get acquainted with playing as a cyborg without clutter
- A cyborg module primarily focused around assisting crew members in the general sense as opposed to strictly through a job

Keeping it simple, the assistant cyborg comes equipped with the following items:
- The standard cyborg flash
- A full-sized fire hydrant for putting out flaming crew members
- An epipen to stabilize crew members in critical condition
- A welder to fix simple machinery and yourself, as well as a weapon if need be
- A crowbar, in case you need to open an unpowered door
- When emagged, has a cyborg e-sword, a straightforward but strong tool originally wielded by the Default Cyborg, perfect for assisting your new friend with whatever his plans are

On top of this, to really promote using this module for assisting the crew, the assistant cyborg gains a 25% speed bonus when being ridden by a living humanoid (so you can't use a corpse to become faster).  This should promote the assistant cyborg to get involved with helping people and for people to be able to utilize it, as it can be used to get around the station at a faster rate than most other common alternatives.  It also allows a malf AI to benefit from their existence by their ability to ferry people to a borging machine quicker than other cyborgs.

Note: if there's any other items anyone can think of which would fit the theme of a "cyborg assistant" (not in the greytiding sense, moreso an actual assistant), let me know.  I'm open to suggestions just as long as it doesn't become the very thing the standard borg got removed for being.

## Why It's Good For The Game

Fills the hole in the lineup that the standard cyborg's removal created, along with giving new borg players an option which promotes learning how everything works while still being useful to the crew.

## Changelog
:cl:
add: Following the removal of standard cyborgs, a new, cheaper module called the Assistant Cyborg has been made public for usage on Nanotrasen space stations.
/:cl:

### CODE REVIEW

**Comment by ishitbyabullet** [comment]
OH NO - NOT THE SILICON TIDE

**Comment by Jared-Fogle** [comment]
> A borg subtype for new players to get acquainted with playing as a cyborg without clutter

Isn't this the point of the service borg?

**Comment by IndieanaJones** [comment]
> Isn't this the point of the service borg?

Seems like a lot of clutter to me just looking at its listed items on the wiki, service borg is more playing bartender and cyborg at the same time moreso than a beginner borg.  Also, while the service borg is sort of a meme who just exists to try and convince humans to get drunk, this module is intended to be actually helpful.

**Comment by wesoda25** [comment]
Wow, you’re re adding standard borgs after coders removed them in seek of hyper-specialized borgs.

**Comment by IndieanaJones** [comment]
> Wow, you’re re adding standard borgs after coders removed them in seek of hyper-specialized borgs.

These borgs are really good are specialized though by being directly focused on human assistance.  Unlike any other borg type, which has a set task it can perform without any need for human interaction, the assistant cyborg exists to assist an enhance a crewmember's experience by getting them around quickly (speed boost and crowbar) and being able to also assist them in a pinch (speed boost and hypospray, along with crowbar in case of power outage).  The whole idea for theme here is a direct human assistant who mostly serves as a convenience without doing so by just having the powers of other borgs.  If you have suggestions on the kit, feel free to speak your mind.

File: code/modules/mob/living/silicon/robot/robot_modules.dm [hunk 1]
```diff
@@ -248,6 +250,19 @@
...
 	return TRUE
 
+/obj/item/robot_module/assistant
+	name = "Assistant"
+	basic_modules = list(
+		/obj/item/assembly/flash/cyborg,
+		/obj/item/reagent_containers/borghypo/epi,
+		/obj/item/weldingtool/largetank/cyborg,
+		/obj/item/crowbar/cyborg,
+		/obj/item/extinguisher)
```
Comment (ATH1909) [inline code/modules/mob/living/silicon/robot/robot_modules.dm:260]: jesus christ, and I thought the PK borg's loadout was anemic

what the hell kind of a borg has only FIVE TOOLS (six if you count the speed boost), with FOUR of them being present in the engiborg module?

the speed boost is interesting, I guess, but it's really janky, especially with how dragging people while you have someone buckled to you as a borg works. tbh, it'd probably work better as just a passive, always-on speedboost.

Comment (ATH1909) [reply code/modules/mob/living/silicon/robot/robot_modules.dm:260]: maybe work on a cargo borg, that sounds more like what you want here.

maybe also let borgs move crates that they're dragging to an empty tile that they click on like humans can? Not being able to do that is currently quite annoying, and probably is the main thing (other than spritework) stopping a cargo borg from being made.

**Comment by IndieanaJones** [comment]
Eh, fuck it for now, I guess.  I tried.

File: code/modules/mob/living/silicon/robot/robot_modules.dm (hunk 1 above)
Comment (IndieanaJones) [reply code/modules/mob/living/silicon/robot/robot_modules.dm:260]: A basic cyborg whose main gimmick is both being a learner borg and moreseo focusing on direct human interaction.  Speedboost is a pretty strong ability, considering what you can do with it.  A borg of this vein is the only way I could see a "plain" borg being specialized, but I have better things to do.

Comment (IndieanaJones) [reply code/modules/mob/living/silicon/robot/robot_modules.dm:260]: Also, the security borg also only had 5 tools by the end of its run.

**Comment by trerri** [comment]
I think this PR is salvageable if you make the riding human require to be sentient, ~~and rename this to Taxi Module.~~

**Pull request closed without merging**